Assignment1_q1_alternative_sol.py

```diff
@@ -0,0 +1,16 @@
+from collections import Counter
+
+# Returns true if strings 's1' and 's2' are permutations of the other, false otherwise
+def is_permutation(s1, s2):
+
+    # We are case-insensitive
+    s1 = s1.lower()
+    s2 = s2.lower()
+
+    # Character count in both strings needs to be the same for them to be permutations
+    return Counter(s1) == Counter(s2)
+
+# Test examples
```

Comment:
Besides these examples, you should actually write automated tests for your code.
Automated tests (using the `unittest` module) should consider examples like the one below, but also focus on corner cases and error cases.